resin/logs.py

```diff
@@ -1,133 +1,191 @@
-from pubnub import Pubnub
 from functools import wraps
-import logging
+import json
+try:  # Python 3 imports
+    from urllib.parse import urljoin
+except ImportError:  # Python 2 imports
+    from urlparse import urljoin
+from collections import defaultdict
+from threading import Thread
+
+from twisted.internet import reactor, ssl
+from twisted.internet.defer import Deferred
+from twisted.internet.protocol import Protocol
+from twisted.web.client import Agent
+from twisted.web.http_headers import Headers
+from twisted.internet.ssl import ClientContextFactory
 
 from .base_request import BaseRequest
+from . import exceptions
 from .models.config import Config
 from .models.device import Device
+from .settings import Settings
 
 
-logging.basicConfig(format='%(levelname)s:%(message)s')
+class WebClientContextFactory(ClientContextFactory):
+    """
+    This is low level class and is not meant to be used by end users directly.
+    """
 
+    def getContext(self, hostname, port):
+        return ClientContextFactory.getContext(self)
 
-def _print_deprecation_warning():
-    logging.warning(" using legacy logging services, this will stop working shortly.\nPlease update to ensure logs are correctly retrieved in future.")
 
+class StreamingParser(Protocol):
+    """
+    This is low level class and is not meant to be used by end users directly.
+    """
 
-# TODO: https://github.com/resin-io/resin-sdk/pull/277/files
-class Logs(object):
+    def __init__(self, callback, error):
+        self.callback = callback
+        self.error = error
+        self.pending = b''
+
+    def dataReceived(self, data):
+        self.pending += data
+
+        lines = self.pending.split(b'\n')
+        self.pending = lines.pop()
+
+        for line in lines:
+            try:
+                if line:
+                    obj = json.loads(line)
+                    self.callback(obj)
+            except Exception as e:
+                if self.error:
+                    self.error(e)
+                    self.transport.stopProducing()
```

Comment:
You should stop producing and close the connection regardless of whether an error callback is registered